orchestration/dagster_orchestration/hca_manage/manage.py

```diff
@@ -231,6 +231,38 @@ def submit_snapshot_request(self, qualifier: Optional[str] = None) -> str:
         logging.info(f"Snapshot creation job id: {response.id}")
         return response.id
 
+    def delete_snapshot(self, snapshot_name: Optional[str] = None, snapshot_id: Optional[str] = None):
+        if snapshot_name and not snapshot_id:
+            response = self.data_repo_client.enumerate_snapshots(filter=snapshot_name)
+            try:
+                snapshot_id = response.items[0].id
+            except IndexError:
+                raise ValueError("The provided snapshot name returned no results.")
+        elif snapshot_id and not snapshot_name:
+            pass  # let snapshot_id argument pass through
```

Comment:
❓ Why is this block needed at all?

Reply:
If it isn't there, then this case would fall under `else` and raise the `ValueError`. I could remove this case, remove the `else`, and make a more targeted `elif` for the value error, but I figured being explicit here that we "do nothing" in this case is more readable for our future selves. Can change if strongly opinionated.